### ISSUE TITLE
Cannot use yang.connector library without installing pyats

### ISSUE THREAD

**Issue by miott** [CiscoTestAutomation/yang#104]
Created a virtualenv and installed yang.connector.  Tried to import a yang.connector class and an error that unicon is missing was reported.  Tried adding unicon and got an error saying pyats is missing.
```
(yang) ~/pyats/install # pip list
Package        Version
-------------- -------
bcrypt         4.0.1
cffi           1.16.0
cryptography   41.0.5
grpcio         1.59.2
lxml           4.9.3
ncclient       0.6.15
paramiko       3.3.1
pip            23.3.1
protobuf       4.25.0
pycparser      2.21
PyNaCl         1.5.0
setuptools     68.2.2
six            1.16.0
wheel          0.41.3
yang.connector 23.10
(yang) ~/pyats/install # python
Python 3.8.16 (default, Dec  7 2022, 01:27:54) 
[Clang 14.0.0 (clang-1400.0.29.202)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from yang.connector import Gnmi
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/miott/pyats/install/yang/lib/python3.8/site-packages/yang/connector/__init__.py", line 22, in <module>
    from .gnmi import Gnmi
  File "/Users/miott/pyats/install/yang/lib/python3.8/site-packages/yang/connector/gnmi.py", line 10, in <module>
    from unicon.sshutils import sshtunnel
ModuleNotFoundError: No module named 'unicon'
>>> from yang.connector import Netconf
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/miott/pyats/install/yang/lib/python3.8/site-packages/yang/connector/__init__.py", line 22, in <module>
    from .gnmi import Gnmi
  File "/Users/miott/pyats/install/yang/lib/python3.8/site-packages/yang/connector/gnmi.py", line 10, in <module>
    from unicon.sshutils import sshtunnel
ModuleNotFoundError: No module named 'unicon'
>>> 
```
Installed unicon and tried again.
```
Installing collected packages: wcwidth, pyyaml, PrettyTable, dill, unicon.plugins, unicon
Successfully installed PrettyTable-3.9.0 dill-0.3.7 pyyaml-6.0.1 unicon-23.10 unicon.plugins-23.10 wcwidth-0.2.9
(yang) ~/pyats/install # python            
Python 3.8.16 (default, Dec  7 2022, 01:27:54) 
[Clang 14.0.0 (clang-1400.0.29.202)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from yang.connector import Netconf
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/miott/pyats/install/yang/lib/python3.8/site-packages/yang/connector/__init__.py", line 24, in <module>
    from .grpc import Grpc
  File "/Users/miott/pyats/install/yang/lib/python3.8/site-packages/yang/connector/grpc/__init__.py", line 6, in <module>
    from pyats.connections import BaseConnection
ModuleNotFoundError: No module named 'pyats.connections'
>>> 
```
Installed yang.connector==23.9 and issue went away.
```
Installing collected packages: yang.connector
  Attempting uninstall: yang.connector
    Found existing installation: yang.connector 23.10
    Uninstalling yang.connector-23.10:
      Successfully uninstalled yang.connector-23.10
Successfully installed yang.connector-23.9
(yang) ~/pyats/install # python                          
Python 3.8.16 (default, Dec  7 2022, 01:27:54) 
[Clang 14.0.0 (clang-1400.0.29.202)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from yang.connector import Netconf
>>> 
```
After investigating, in 23.10, "grpc" was added which has imports required from pyats and the connector/gnmi.py added an import from unicon.

**Comment by SohanTirpude** [comment]
Hello @miott,

Currently I am looking into this issue, kindly let me check with the team on this.

**Comment by miott** [comment]
> Hello @miott,
> 
> Currently I am looking into this issue, kindly let me check with the team on this.

I have a pull request open for the fix. #107

**Comment by SohanTirpude** [comment]
Hello @miott, thank you for letting me know about the PR.

**Comment by SohanTirpude** [comment]
Hello @miott, your PR got required numbers of approvals, so kindly merge the PR.

**Comment by SohanTirpude** [comment]
Hello @miott,

Your PR got merged and it seems the fix now will be available in 23.12 hence I will be closing this ticket. Thank you very much for handling this issue by yourself.
